Cinema system/Cinema system/Services/CinemaService.cs

```diff
@@ -0,0 +1,151 @@
+﻿using CinemaSystem.Interfaces;
+using CinemaSystem.Models;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel.DataAnnotations;
+
+namespace CinemaSystem.Services
+{
+    public class CinemaService : ICinemaService
+    {
+
+        private readonly CinemaDBContext _db;
+
+        public CinemaService(CinemaDBContext db)
+        {
+            _db = db;
+        }
+
+        public Errors IsCinemaExists(int cinemaId)
+        {
+            Errors errors = new Errors();
+
+            if (!_db.Cinemas.Any(c => (c.Id == cinemaId && !c.IsDeleted)))
+            {
+                errors.IsCorrect = false;
+                errors.Description.Add("This cinema does not exists.");
+                return errors;
+            }
+
+            errors.IsCorrect = true;
+            return errors;
+        }
+
+        public Errors IsCinemaExistsAndValid(CinemaView cinema)
+        {
+            Errors errors = new Errors();
+            var results = new List<ValidationResult>();
+            var context = new ValidationContext(cinema);
+
+            if (!Validator.TryValidateObject(cinema, context, results, true))
+            {
+                errors.IsCorrect = false;
+                foreach (var error in results)
+                {
+                    errors.Description.Add(error.ErrorMessage);
+                }
+                return errors;
+            }
+            else
+            {
+                if (_db.Cinemas.Any(c => (c.Title == cinema.Title && c.City == cinema.City && !c.IsDeleted)))
+                {
+                    errors.IsCorrect = false;
+                    errors.Description.Add("This cinema is already exists.");
+                    return errors;
+                }
+            }
+
+            errors.IsCorrect = true;
+            return errors;
+        }
+
+        public Errors IsCinemaDataValid (CinemaView cinema)
+        {
+            Errors errors = new Errors();
+
+            var results = new List<ValidationResult>();
+            var context = new ValidationContext(cinema);
+
+            if (!Validator.TryValidateObject(cinema, context, results, true))
+            {
+                errors.IsCorrect = false;
+                foreach (var error in results)
+                {
+                    errors.Description.Add(error.ErrorMessage);
+                }
+            }
+            else
+            {
+                errors.IsCorrect = true;
+            }
+
+            return errors;
+        }
+        
+
+        private Cinema GenerateNewCinema(CinemaView cinema)
```

Comment:
Do you really need this method now? What is its purpose besides creating one object from the other?